<?xml version="1.0"?>
<launch>

  <arg name="rviz" default="true" />
  <arg name="scanPeriod" default="0.1" />

  <!-- loam -->
  <node pkg="loam_velodyne" type="multiScanRegistration" name="multiScanRegistration" output="screen" respawn="true">
    <param name="lidar" value="HDL-64E" /> <!-- options: VLP-16  HDL-32  HDL-64E -->
    <param name="scanPeriod" value="$(arg scanPeriod)" />
    <remap from="/multi_scan_points" to="/velodyne_points_down"/>
    <!--remap from="/imu/data" to="/kitti/oxts/imu"/-->
  </node>
  <node pkg="loam_velodyne" type="laserOdometry" name="laserOdometry" output="screen" respawn="true">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>
  <node pkg="loam_velodyne" type="laserMapping" name="laserMapping" output="screen" respawn="true">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>
  <node pkg="loam_velodyne" type="transformMaintenance" name="transformMaintenance" output="screen" respawn="true"/>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find loam_velodyne)/rviz_cfg/loam_velodyne.rviz" />
  </group>

  <!-- downsample the input points -->
  <node pkg="loam_velodyne" type="downsample_input_points" name="downsample_input_points" output="screen" respawn="true">
    <param name="leafsize" value="0.05" />
    <remap from="/velodyne_points_orig" to="/velodyne_points"/>
  </node>
  
  <node pkg="loam_velodyne" type="save_odometry" name="save_odometry" output="screen" respawn="true" />
   

  <!--node pkg="tf" type="static_transform_publisher" name="static_tf" args="3.0 0.0 0.0 0.0 0.0 1.57 velodyne_link camera_init 20"/--> 

</launch>
